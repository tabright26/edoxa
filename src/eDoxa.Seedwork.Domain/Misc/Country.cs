// Filename: Country.cs
// Date Created: 2019-10-12
// 
// ================================================
// Copyright © 2019, eDoxa. All rights reserved.

using System.ComponentModel;

namespace eDoxa.Seedwork.Domain.Misc
{
    [TypeConverter(typeof(EnumerationTypeConverter))]
    public sealed class Country : Enumeration<Country>
    {
        public static readonly Country Canada = new Country(
            1,
            "CA",
            "CAN",
            "1");
        public static readonly Country UnitedStates = new Country(
            1 << 1,
            "US",
            "USA",
            "1");

        public Country()
        {
            ThreeDigitIso = string.Empty;
            CallingCode = string.Empty;
        }

        private Country(
            int value,
            string name,
            string threeDigitIso,
            string callingCode
        ) : base(value, name)
        {
            ThreeDigitIso = threeDigitIso;
            CallingCode = callingCode;
        }

        public string TwoDigitIso => Name;

        public string ThreeDigitIso { get; }

        public string CallingCode { get; }
    }
}
