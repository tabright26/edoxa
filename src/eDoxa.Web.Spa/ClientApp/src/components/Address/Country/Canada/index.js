import Canada from "./Canada";

export default Canada;
