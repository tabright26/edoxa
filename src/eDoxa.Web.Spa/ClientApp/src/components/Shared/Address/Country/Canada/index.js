import CanadaCountry from "./Canada";

export default CanadaCountry;
