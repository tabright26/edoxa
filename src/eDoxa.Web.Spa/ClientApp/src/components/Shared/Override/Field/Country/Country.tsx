import React from "react";
import { Field } from "redux-form";
import Input from "components/Shared/Override/Input";
import { countries } from "components/Shared/Localization/countries";

const CountryField = () => (
  <Field name="country" type="select" component={Input.Select}>
    {countries.map(country => (
      <option value={country.twoDigitIso}>{country.name}</option>
    ))}
  </Field>
);

export default CountryField;
