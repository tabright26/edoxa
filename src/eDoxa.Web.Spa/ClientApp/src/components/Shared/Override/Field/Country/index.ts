import CountryField from "./Country";

export default CountryField;
