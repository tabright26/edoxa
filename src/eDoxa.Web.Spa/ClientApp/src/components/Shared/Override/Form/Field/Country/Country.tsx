import React from "react";
import { Field } from "redux-form";
import Input from "components/Shared/Override/Input";
import { countries } from "utils/localize/countries";

const CountryField = ({ disabled = false }) => (
  <Field name="country" type="select" component={Input.Select} disabled={disabled}>
    {countries.map((country, index) => (
      <option key={index} value={country.twoDigitIso}>
        {country.name}
      </option>
    ))}
  </Field>
);

export default CountryField;
