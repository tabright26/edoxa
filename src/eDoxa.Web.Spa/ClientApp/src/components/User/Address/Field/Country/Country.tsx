import React, { FunctionComponent, SyntheticEvent } from "react";
import { Field } from "redux-form";
import Input from "components/Shared/Input";
import { compose } from "recompose";
import { connect, MapStateToProps } from "react-redux";
import { RootState } from "store/types";
import { CountryOptions } from "types";
import { FormGroup, Label } from "reactstrap";

interface OwnProps {
  label?: string;
  placeholder: string;
  disabled?: boolean;
  onChange?: (event: SyntheticEvent) => void;
}

interface StateProps {
  countries: CountryOptions[];
}

type InnerProps = StateProps;

type OutterProps = OwnProps;

type Props = InnerProps & OutterProps;

const CountryField: FunctionComponent<Props> = ({
  countries,
  disabled = false,
  label = null,
  placeholder,
  onChange = null
}) => (
  <FormGroup>
    {label && <Label>{label}</Label>}
    <Field
      name="country"
      type="select"
      placeholder={placeholder}
      component={Input.Select}
      disabled={disabled}
      onChange={onChange}
    >
      {countries.map((country, index) => (
        <option key={index} value={country.twoIso}>
          {country.name}
        </option>
      ))}
    </Field>
  </FormGroup>
);

const mapStateToProps: MapStateToProps<
  StateProps,
  OwnProps,
  RootState
> = state => {
  return {
    countries: state.static.identity.data.addressBook.countries
  };
};

const enhance = compose<InnerProps, OutterProps>(connect(mapStateToProps));

export default enhance(CountryField);
