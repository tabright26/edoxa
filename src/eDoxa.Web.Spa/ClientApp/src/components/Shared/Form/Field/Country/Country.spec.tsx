import React from "react";
import renderer from "react-test-renderer";
import { Form, reduxForm } from "redux-form";
import { Provider } from "react-redux";
import Country from "./Country";

it("renders without crashing", () => {
  //Arrange
  const store: any = {
    getState: () => {},
    dispatch: action => {},
    subscribe: () => {}
  };

  const FormWrapper = () => (
    <Form onSubmit={() => {}}>
      <Country />
    </Form>
  );

  const CustomForm = reduxForm({ form: "TEST_FORM" })(FormWrapper);

  // Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <CustomForm />
      </Provider>
    )
    .toJSON();

  //Assert
  expect(tree).toMatchSnapshot();
});
