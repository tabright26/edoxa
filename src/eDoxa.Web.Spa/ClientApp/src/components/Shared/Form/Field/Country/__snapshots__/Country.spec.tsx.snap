// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<form>
  <select
    className="form-control-sm form-control"
    disabled={false}
    name="country"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    value=""
  >
    <option
      value="CA"
    >
      Canada
    </option>
    <option
      value="US"
    >
      United States
    </option>
  </select>
</form>
`;
